internal/compiler/object_tree.rs

```diff
@@ -838,25 +888,37 @@ impl Element {
             for prop_name_token in anim.QualifiedName() {
                 match QualifiedTypeName::from_node(prop_name_token.clone()).members.as_slice() {
                     [unresolved_prop_name] => {
-                        let PropertyLookupResult { resolved_name, property_type } =
-                            r.lookup_property(unresolved_prop_name);
+                        let lookup_result = r.lookup_property(unresolved_prop_name);
+                        let valid_assign = lookup_result.is_valid_for_assignment();
                         if let Some(anim_element) = animation_element_from_node(
                             &anim,
                             &prop_name_token,
-                            property_type,
+                            lookup_result.property_type,
                             diag,
                             tr,
                         ) {
-                            if unresolved_prop_name != resolved_name.as_ref() {
+                            if !valid_assign {
+                                diag.push_error(
+                                    format!(
+                                        "animating {} property '{}'",
```

Comment:
I think that it might be better if the error message not only says what is attempted ("animateing inout property") but also that it is not possible/allowed. So for example "Cannot animate inout property".